lib/oban/job.ex

```diff
@@ -293,10 +314,29 @@ defmodule Oban.Job do
   def valid_unique_opt?({:states, [_ | _] = states}), do: states -- states() == []
   def valid_unique_opt?(_option), do: false
 
+  @time_units ~w(
+    second
+    seconds
+    minute
+    minutes
+    hour
+    hours
+    day
+    days
+    week
+    weeks
+    month
+    months
+  )a
+
+  defguardp is_timestampable(value)
+            when is_integer(value) or
```

Comment:
Lovely use of a private guard. Something I noticed, between this and the property test, is that we're accepting negative values. Enforcing positive integers now would be a backward-incompatible change, but it struck me.

Reply:
I noticed that too.  The Stager test makes use of the ability to schedule jobs in the past.